packit/upstream.py

```diff
@@ -1103,7 +1103,11 @@ def _add_link_to_archive_from_specdir_if_needed(self, archive_path: Path) -> Non
 
         if archive_path.parent.absolute() != absolute_specfile_dir:
             archive_in_spec_dir = absolute_specfile_dir / archive_path.name
-            relative_archive_path = archive_path.relative_to(absolute_specfile_dir)
+
+            # Path.relative_to can ValueError when specfile_dir is in a subdir and
+            # archive path is in the project's root: https://github.com/packit/packit/issues/1621
+            # therefore we are using os.path.relpath here - it's a different implementation
```

Comment:
Please add this quite from the docs:

> [PurePath.relative_to()](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.relative_to) requires self to be the subpath of the argument, but [os.path.relpath()](https://docs.python.org/3/library/os.path.html#os.path.relpath) does not.

Explains the reason better.

Reply:
ha, I missed that one, thanks!